### PR TITLE
Adding elevator

### PR DESCRIPTION
**Pull request by evan-gan**
Still need to adjust some values with comments next to them, otherwise should be good.

### CODE REVIEW

File: src/main/java/frc/robot/subsystems/Elevator.java [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+
+    // For initial testing, these should be very slow.
+    // We can update them as we get more confidence.
+    private static final double ELEVATOR_MAX_VEL_METER_PER_SEC = Units.inchesToMeters(100.0); //TODO: Find actual value
+
+    // private static final double ELEVATOR_MAX_ACC_METER_PER_SEC_SQ = Units.inchesToMeters(50.0);
+
+    private static final double ELEVATOR_MAX_ACC_METER_PER_SEC_SQ = Units.inchesToMeters(100.0); //TODO: Find actual value
+
+    private static final double ELEVATOR_METER_PER_REVOLUTION = Units.inchesToMeters((1.504*Math.PI)/4); //May be swiched to 1:5, NEED TO CHECK
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/subsystems/Elevator.java:33]: Put in the gear ratio multiplications.

File: src/main/java/frc/robot/subsystems/Elevator.java [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+import edu.wpi.first.math.trajectory.TrapezoidProfile;
+import edu.wpi.first.math.util.Units;
+import edu.wpi.first.wpilibj.AnalogPotentiometer;
+import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard;
+import edu.wpi.first.wpilibj2.command.TrapezoidProfileSubsystem;
+import frc.robot.Constants;
+
+public class Elevator extends TrapezoidProfileSubsystem {
+
+    private static final double ELEVATOR_MAX_LENGTH = Units.inchesToMeters(22.0);
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/subsystems/Elevator.java:20]: Add units to make this ELEVATOR_MAX_LENGTH_METERS

File: src/main/java/frc/robot/subsystems/Elevator.java [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    private static final double ELEVATOR_MIN_LENGTH = Units.inchesToMeters(0.0);
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/subsystems/Elevator.java:21]: Add units to make this ELEVATOR_MIN_LENGTH_METERS

File: src/main/java/frc/robot/subsystems/Elevator.java [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    //Offset for command
+    public static final double REACHER_OFFSET_TOLERANCE_METERS = 1; //TODO: pick value, used in the command SetElevator length
+
+    // Define the motor and encoders
+    private final CANSparkMax m_motor;
+    private final RelativeEncoder m_encoder;
+
+    //initializing Potentiometer
+    private final int POTENTIOMETER_CHANNEL = 2; //TODO: Update with actual value
+    private final double POTENTIOMETER_RANGE = -2.625; // meters, the string potentiometer on takes in range in integers TODO: update to correct value
```
Comment (coachFitz0807) [inline src/main/java/frc/robot/subsystems/Elevator.java:53]: Add units to make this POTENTIOMETER_RANGE_METERS

**Review by coachFitz0807 — COMMENTED** [review]
Use the Rename Symbol in the right-click pop-p to rename the symbols so you only have to do it once.

**Review by Weq511762338 — APPROVED** [review]

**Pull request merged**